### ISSUE TITLE
WiX 3.14 with arm64 support release

### ISSUE THREAD

**Issue by Alovchin91** [wixtoolset/issues#6269]
Hi,

Is there any chance WiX 3.14 release with arm64 support will see the light of day?

I know that some projects successfully use pre-release builds, but I also see many projects skipping building MSI for Windows on arm64 because there is no stable WiX build for that.

What is your plan around it? Thanks.

**Comment by rseanhall** [comment]
I'll just throw this out there - Microsoft is using the latest version of WiX 3.14 to distribute .NET 5. https://github.com/dotnet/installer/blob/12b9c8d67e91d83ece3ca70e6899e29f7f3a8f10/src/redist/targets/GenerateMSIs.targets#L17

**Comment by barnson** [comment]
If there is an "official" release of WiX v3.14, it will come after WiX v4.0. In the meantime, the weekly builds are available to enjoy.

**Comment by Alovchin91** [comment]
That's unfortunate.

Would it be possible to clearly communicate this in the readme or on the website? So people don't wait for arm64 supporting release.

**Comment by robmen** [comment]
@Alovchin91 there isn't a final decision on WiX v3.14's final release. It is correctly marked on the website as a Development Release. We'll update its status when we decide. Our focus is on WiX v4.0.

**Comment by Alovchin91** [comment]
@robmen  Thank you for the clarification! Please keep in mind that some projects are waiting for an "official" release, skipping the "development" one unfortunately.

**Comment by BMurri** [comment]
@Alovchin91 we understand, and that is completely appropriate. Everyone (including Microsoft's Net 5 team) using builds we've marked "development" have guidance (based on standard conventions both inside and outside of the OSS community) that they should be closely monitoring our public channels of communication, staying as close as possible to the latest builds, and anticipating source code changes from build-to-build, as well as feeding us bug reports (and of course we always welcome help, especially help that will permit us to release sooner rather than later).

Not every project has the resources to do that. It's up to each project that consumes our toolset to make those judgements based on their own resources and needs.

We aren't promising dates at this point, but steady progress is being made, and having been on several previous product pushes, this one feels close.

**Comment by rhuijben** [comment]
Without a proper release it is unlikely that GitHub workflows is going to install this version in their default buildchain, so delivering ARM64 software will remain hard until we have some Wix release that supports this.

As @A-Ovchinnikov-mx mentioned this is holding back Subversion related releases for ARM64.

**Comment by robmen** [comment]
@rhuijben I understand everything you are saying there. Know that producing another production v3.x release is a distant--very, very, very distant--second priority to completing WiX v4.0.

WiX v4.0 is designed to integrate seamlessly into CI/CD so that you/we are not dependent on the CI/CD providers updating their build images. The tooling is much easier to acquire. `v4-preview.0` already demonstrates the `dotnet tool` route, as per: https://robmensching.com/blog/posts/2021/5/17/wix-toolset-v4-preview.0/. More options come in `-preview.1`.

At this point (assuming you're looking for purely free solutions) you'd help us more by starting to adopt WiX v4 (which has ARM64 support). Your adoption will help flush out bugs in `-preview.0` now and `-preview.1` later. That helps us finish WiX v4 faster.

tl;dr WiX v4 will release before there is another v3.x build.

**Comment by Alovchin91** [comment]
Wix Toolset currently blocks quite a bunch of open-source projects depending on it. Given a very slow progress on Wix 4 (I understand that it is a huge project BTW), I believe the decision to not release 3.x with ARM64 support should be reconsidered.

**Comment by barnson** [comment]
Nobody is blocked from using WiX v3.14.

**Comment by Alovchin91** [comment]
@barnson Projects are blocked by the fact that it's not officially released. Projects have policies. So yes, they're blocked.

**Comment by Alovchin91** [comment]
If you're confident about 3.14 quality, why not just proclaim it a release? That would unblock everyone.

**Comment by robmen** [comment]
@Alovchin91 we're not confident that v3.14 should be declared production quality for the full breadth of functionality available in the WIX Toolset. @barnson's point is that if v3.14 works for your scenarios then you can use it (some have been successful with this approach).

Additionally, there are breaking changes in v3.14 to support the migration to v4.0 and more are possible (likely?) as we finish v4.0. That is actually the primary focus of v3.14, supporting the migration to v4.0.

We _could_ create a new plan around a new WiX v3.x release but that would distract us from v4.0 even longer. Not just creating and testing the new release (and given you are talking about ARM64 support probably doing additional work since ARM64 support is not complete in WiX v3.x) but supporting yet another v3.x on top of all the other releases.

Finally, I don't feel like spending my free time managing a new release (*especially* given the amount of work that is dependent on me in v4.0). Doing another v3.x release would only add to the pressure I feel around getting v4.0 complete. I will go the extra mile for customers and contributors. But in my free time, I need to feel like I'm doing something personally productive.

**Comment by Alovchin91** [comment]
@robmen I understand your concern. Wix 4 is a fantastic idea and clearly everybody would love it to happen.

But it's almost mythical, with no release date and no roadmap to my knowledge. It won't suffer from some more delay.

At the same time, more and more people need Windows arm64 support here and now. I believe the focus could change when there's a need.

**Comment by robmen** [comment]
> It won't suffer from some more delay.

Maybe. But I would.

**Comment by Alovchin91** [comment]
Sorry to hear that 😞

Maybe at least a note on the website stating that "if your scenario works, 3.14 is the way to go for Windows arm64 support" would suffice?

In particular, what's important is _creating_ MSIs that support Windows arm64, not _running_ Wix tools natively on it.

**Comment by niemyjski** [comment]
I wish 3.14 would be released officially so ci build images with VS2022 (like github actions) could have VS2022 support

> VS2022DEVENV | Full path to devenv.exe for Visual Studio 2022 if it is installed on the system. This property is available starting with WiX v3.14.

**Comment by barnson** [comment]
Problem solved: https://wixtoolset.org/docs/v3/msbuild/daily_builds/
